evm/v0.5/src/helpers.ts

```diff
@@ -603,24 +665,183 @@ export function generateSAID(sa: ServiceAgreement): Hash {
     SERVICE_AGREEMENT_TYPES,
     serviceAgreementValues(sa),
   )
 }
 
+/**
+ * Generates the bytes to call and execute the oracleRequest() function
+ * @param sAID The service agreement ID
+ * @param callbackAddr The callback contract address
+ * @param callbackFunctionId The callback function id
+ * @param nonce The nonce
+ * @param data optional data
+ */
+export function executeServiceAgreementBytes(
+  sAID: string,
+  callbackAddr: string,
+  callbackFunctionId: any,
+  nonce: utils.BigNumberish,
+  data?: any,
+): any {
+  const types = [
+    'address',
+    'uint256',
+    'bytes32',
+    'address',
+    'bytes4',
+    'uint256',
+    'uint256',
+    'bytes',
+  ]
+  const values = [
+    ethers.constants.AddressZero,
+    0,
+    sAID,
+    callbackAddr,
+    callbackFunctionId,
+    nonce,
+    1,
+    data || '0x',
+  ]
+  const encodedArgs = utils.defaultAbiCoder.encode(types, values)
+  const funcSelector = encodeFunctionSignature(
+    'oracleRequest(address,uint256,bytes32,address,bytes4,uint256,uint256,bytes)',
+  )
+  return utils.concat([funcSelector, encodedArgs])
+}
+
+/**
+ * Generates the oracle signatures on a ServiceAgreement
+ * @param serviceAgreement The service agreement to sign
+ * @param signers The list oracles that will sign the service agreement
+ */
+export async function generateOracleSignatures(
+  serviceAgreement: ServiceAgreement,
+  signers: ethers.Wallet[],
+): Promise<OracleSignatures> {
+  // require signing wallets match oracle addresses in SA
+  assert.deepEqual(
+    serviceAgreement.oracles,
+    signers.map(w => w.address),
+    'oracle wallets do not match list of oracles on service agreement',
+  )
+
+  const sAID = generateSAID(serviceAgreement)
+  const signatures = []
+
+  for (let i = 0; i < signers.length; i++) {
+    const signer = signers[i]
+    const oracle = serviceAgreement.oracles[i]
+    const oracleSignature = await personalSign(sAID, signer)
+    const requestDigestAddr = recoverPersonalSignature(sAID, oracleSignature)
+    assert.equal(oracle, requestDigestAddr)
+    signatures.push(oracleSignature)
+  }
+
+  return combineOracleSignatures(signatures)
+}
+
+/**
+ * Combine v, r, and s params of multiple signatures into format expected by contracts
+ * @param signatures The list of signatures to combine
+ */
+export function combineOracleSignatures(
+  signatures: Signature[],
+): OracleSignatures {
+  return {
+    vs: signatures.map(os => os.v),
+    rs: signatures.map(os => os.r),
+    ss: signatures.map(os => os.s),
+  }
+}
+
+/**
+ * Signs a message
+ * @param message The message to sign signed
+ * @param wallet The wallet of the signer
+ */
+export async function personalSign(
+  message: any,
+  wallet: ethers.Wallet,
+): Promise<Signature> {
+  const eMsg =
+    `Message ${message} is not a recognized representation of a ` +
+    'byte array. (Can be Buffer, BigNumber, Uint8Array, 0x-prepended ' +
+    'hexadecimal string.)'
+  assert(isByteRepresentation(message), eMsg)
```

Comment:
Follow up with type restriction, we wont need the `isByteRepresentation` function either if we do so.